easybuild/tools/options.py

```diff
@@ -1581,6 +1590,20 @@ def set_up_configuration(args=None, logfile=None, testing=False, silent=False):
         'try_to_generate': try_to_generate,
         'valid_stops': [x[0] for x in EasyBlock.get_steps()],
     }
+
+    # Remove existing singletons if reconfigure==True (allows reconfiguration when looping over EasyStack items)
+    if reconfigure:
```

Comment:
Also for this new aspect of `set_up_configuration` we should make sure there's a test, to verify that this is indeed working as expected...

It seems like there's no dedicated test for `set_up_configuration` at all yet, that belongs in `easybuild.tools.options`...

Reply:
Done in https://github.com/easybuilders/easybuild-framework/pull/4057/commits/b11f346604599eea8e1aec6f8f9be4c9dea5382c